internal/services/executor/executor.go

```diff
@@ -1517,3 +1521,36 @@ func (e *Executor) Run(ctx context.Context) error {
 
 	return nil
 }
+
+func (e *Executor) tasksTimeoutCleanerLoop(ctx context.Context) {
+	for {
+		e.log.Debug().Msgf("tasksTimeoutCleaner")
+
+		e.tasksTimeoutCleaner(ctx)
+
+		sleepCh := time.NewTimer(2 * time.Second).C
+		select {
+		case <-ctx.Done():
+			return
+		case <-sleepCh:
+		}
+	}
+}
+
+func (e *Executor) tasksTimeoutCleaner(ctx context.Context) {
+	for _, rtID := range e.runningTasks.ids() {
+		rt, ok := e.runningTasks.get(rtID)
+		if !ok {
+			continue
+		}
+
+		rt.Lock()
+		if rt.et.Spec.TaskTimeoutInterval != 0 && time.Since(*rt.et.Status.StartTime) > rt.et.Spec.TaskTimeoutInterval {
+
+			rt.et.Spec.Timeoutd = true
```

Comment:
I meant to add a Timedout field to `runningTask`, then in `executeTask` when `et.Status.Phase` is updated you should check this value to choosed between failed or timedout.